Refresh.GameServer/Services/VanillaHashChecker.cs

```diff
@@ -0,0 +1,34 @@
+using Refresh.Common.Helpers;
+using Refresh.GameServer.Resources;
+
+namespace Refresh.GameServer.Services;
+
+public static class VanillaHashChecker
+{
+    // A hashset of all the vanilla hashes.
+    // This would be a HashSet<string> since we use strings for hashes everywhere else,
+    // but the perf was like 2x worse and it took 4x the memory,
+    // even with the overhead of the string -> byte[] conversion, each check is cheaper this way.
+    private static readonly HashSet<byte[]> VanillaHashes = new(new ByteArrayComparer());
+
+    static VanillaHashChecker()
+    {
+        using Stream stream = ResourceHelper.StreamFromResource("Refresh.GameServer.Resources.all_hashes.txt");
+        
+        StreamReader reader = new(stream);
+
+        //Read all the lines
+        while (reader.ReadLine() is {} line)
+        {
+            //Parse out and add each line to the set
+            VanillaHashes.Add(HexHelper.HexStringToBytes(line));
+        }
```

Comment:
This is a micro-optimization for sure, but this makes me wonder if it's a good idea to store the hashes in binary, and then just read the bytes directly into `VanillaHashes` so we can skip this conversion.

I guess the drawback is that the file wouldn't be human readable? I'm not sure if that's really that bad though.

Reply:
I switched it to a `FrozenSet`, thanks for the reminder of its existance. I think if this does end up being a problem in the future, we can switch to a binary based format. For reference, the generation code for the file is present [here](https://github.com/Beyley/zishzingers/commit/c3a99ab7c084ceb34a67d9890c778f20d1cb626b)